### PR TITLE
Warn if no audio device is found when compiled with 'video-ffmpeg' option

### PR DESCRIPTION
**Pull request by bu5hm4nn**
Screenshot 2024-03-22 at 13 15 12

### CODE REVIEW

**Comment by mikedilger** [comment]
Is that dismissable?  I know the user _should_ recompile, but they might not be able to at the moment.

We also have a status message area in the sidebar, so now there are two ways to get status messages.  I actually like this bar on the top, if it were dismissable and all sidebar messages went there instead.

**Comment by bu5hm4nn** [comment]
I made it non-dismissable for this case as I see it as a broken compile. Either you want video with sound or you don't want video IMO.

**Comment by bu5hm4nn** [comment]
UPDATE: Message is now dismissable (stored in RAM, so good for one gossip session)

**Comment by dtonon** [comment]
@bu5hm4nn for this type of notification, use the new color `amber_400` (#FBBF24) from the palette for the background, with a neutral_950 for the text. If possible I would put the "Dismiss message" on  a new line, right aligned.

**Comment by mikedilger** [comment]
This is good by me, waiting for changes dtonon suggested

**Comment by bu5hm4nn** [comment]
Completed and updated screenshot

**Comment by bu5hm4nn** [comment]
@dtonon is it how you thought? remember we will improve these things once we have pending, we just want to help people who have tried to compile with video support and don't understand why they don't have sound

**Comment by dtonon** [comment]
@bu5hm4nn I retouched the style:

Screenshot 2024-03-22 alle 23 45 40
<br/><br/>

But there is a problem: the alert is visible only if the top area is opened (so if someone does not decrypt the key, he will basically never see it) and it is _never_ visible if the user has the compose area at the bottom (the default).

PS: I think it is better to _add_ the review instead of replace the original post, so we can follow the design evolution and also refer to older versions

File: gossip-bin/src/ui/mod.rs [hunk 1]
```diff
@@ -1367,21 +1372,48 @@ impl eframe::App for GossipUi {
...
+                    right: 30.0,
+                    top: 20.0,
+                    bottom: 20.0,
                 }),
             )
             .resizable(true)
             .show_animated(
                 ctx,
-                self.show_post_area_fn() && read_setting!(posting_area_at_top),
+                (self.show_post_area_fn() && read_setting!(posting_area_at_top)) || has_warning,
```
Comment (bu5hm4nn) [inline gossip-bin/src/ui/mod.rs:1398]: The OR with `has_warning` here will make sure that the top sidebar is open when the warning is active.

**Comment by bu5hm4nn** [comment]
@dtonon You changed the inner padding of the entire top-sidepanel. That also changed how the compose area is displayed:
Screenshot 2024-03-22 at 17 28 13

File: gossip-bin/src/ui/widgets/mod.rs [hunk 2]
```diff
@@ -166,6 +167,34 @@ pub(super) fn set_important_button_visuals(ui: &mut Ui, app: &GossipUi) {
...
 }
 
+#[allow(dead_code)]
+pub(crate) fn warning_frame<R>(
+    ui: &mut Ui,
+    app: &mut GossipUi,
+    inner: impl FnOnce(&mut Ui, &mut GossipUi) -> R,
+) -> R {
+    egui::Frame::none()
+        .outer_margin(egui::Margin {
```
Comment (bu5hm4nn) [inline gossip-bin/src/ui/widgets/mod.rs:177]: `outer_margin` and `inner_margin` here can be used to change only the warning box

**Comment by dtonon** [comment]
@bu5hm4nn forced pushed a fix.
We need to add a right margin as proposed here: https://github.com/mikedilger/gossip/issues/687

But we still have a glitch with the bottom bar:

Screenshot 2024-03-25 alle 20 58 08
<br/><br/>

It's ok to show it always at the top, but we should not show the duplicate unlock area.

**Comment by bu5hm4nn** [comment]
https://github.com/mikedilger/gossip/assets/3328670/c3b76fe6-c1ce-444a-ab9e-0e637fbbc854

**Pull request merged**